src/OpenTelemetry.Instrumentation.Hangfire/Implementation/HangfireInstrumentationJobFilterAttribute.cs

```diff
@@ -30,8 +31,12 @@ public void OnPerforming(PerformingContext performingContext)
                 return;
             }
 
+            var traceParent = performingContext.GetJobParameter<string>(HangfireInstrumentationConstants.TraceParentKey);
+            var traceState = performingContext.GetJobParameter<string>(HangfireInstrumentationConstants.TraceStateKey);
+            ActivityContext.TryParse(traceParent, traceState, out var parentContext);
```

Comment:
it is not recommended to do this by-hand, as there is a feature to achieve context propagation/extraction : https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Api#context-propagation

Reply:
Hey! Thanks for the comment, I started implementing the change and it was quite easy to do injection of activity context in `OnCreating` method.

```c#
ActivityContext contextToInject = default;
if (Activity.Current is not null)
{
    contextToInject = Activity.Current.Context;
}

Propagator.Inject(new PropagationContext(contextToInject, Baggage.Current), creatingContext.Parameters, InjectActivityProperties);

private static void InjectActivityProperties(IDictionary<string, object> jobParams, string key, string value)
{
    jobParams[key] = value;
}
```

The problem is that in `OnPerforming` the `PerformingContext` is not directly exposing `Parameters` dictionary. Instead if you want to get any of the params you have to call `GetJobParameter<T>` which will hit the underlying store (db, redis...whatever else). 

So I'm not really sure how to continue? Any ideas?

Reply:
I tried using `Fields` from propagator instance to enumerate over them and get all params from db but that didn't work..

<img width="808" alt="Screenshot 2022-06-13 at 19 38 32" src="https://user-images.githubusercontent.com/100698932/173412313-3aac7651-ad21-4ab0-aa30-104224321d82.png">

Reply:
Lastly..it raises another question. If it actually has to go and get the param from the db...won't that be inefficient? Let's say we have 10 params in future, that's 10 separate calls to the database 🤔

Reply:
Maybe could you use a Dictionary<string,string> for the context and put it in a single JobParameter ?

Reply:
> Maybe could you use a Dictionary<string,string> for the context and put it in a single JobParameter ?

@fred2u What a simple and great idea :D I don't know why I didn't think of it. 

I updated code with your suggestion, please take a look. @cijothomas